lib/pbench/server/api/__init__.py

```diff
@@ -52,14 +67,17 @@ def create_app():
     api = Api(app)
 
     app.logger = get_pbench_logger(__name__, config)
-
     app.config_server = config.conf["pbench-server"]
+    app.config_elasticsearch = config.conf["elasticsearch"]
+    app.config_graphql = config.conf["graphql"]
 
     prdp = app.config_server.get("pbench-receive-dir-prefix")
     try:
         upload_directory = Path(f"{prdp}-002").resolve(strict=True)
-    except Exception:
-        app.logger.error("Missing config variable for pbench-receive-dir-prefix")
+    except Exception as e:
+        app.logger.error(
+            "Missing config variable for pbench-receive-dir-prefix, %s", e.__class__
```

Comment:
Seems like we should check if `prdp` received an empty value from the configuration, and emit the missing configuration variable log message without the exception class.

Then, we could resolve the `Path`, catching `FileNotFoundError` if the path is missing, and then using `app.logger.exception()` for any other exception raised.

Thoughts?

Reply:
Indeed, the exception can be narrowed down before we can call the broader Exception class